src/mono/mono/metadata/bundled-resources.c

```diff
@@ -0,0 +1,196 @@
+// most likely to happen during runtime shutdown.
+//
+
+void
+mono_bundled_resources_free (void)
+{
+	g_hash_table_destroy (bundled_resources);
+	bundled_resources = NULL;
+
+	bundle_contains_assemblies = false;
+	bundle_contains_satellite_assemblies = false;
+}
+
+//---------------------------------------------------------------------------------------
+//
+// mono_bundled_resources_free_bundled_resource_func frees all memory allocated for a
+// particular bundled resource. A MonoBundled*Resource should set the underlying
+// MonoBundledResource's free_bundled_resource_func to this if it is being dynamically
+// allocated (i.e. through old bundling api's).
+//
+// Through the old bundling api's, allocations occur for the MonoBundled*Resource.
+// In addition, MonoBundledSatelliteAssemblyResource's allocate for the id.
+//
+
+void
+mono_bundled_resources_free_bundled_resource_func (void *resource)
+{
+	MonoBundledResource *bundled_resource = (MonoBundledResource *)resource;
+
+	if (bundled_resource->type == MONO_BUNDLED_SATELLITE_ASSEMBLY)
+		free ((void *)bundled_resource->id);
+
+	free (resource);
+}
+
+//---------------------------------------------------------------------------------------
+//
+// mono_bundled_resources_value_destroy_func frees the memory allocated by the hashtable's
+// MonoBundled*Resource by invoking its underlying free_bundled_resource_func when possible.
+//
+
+static void
+mono_bundled_resources_value_destroy_func (void *resource)
+{
+	MonoBundledResource *value = (MonoBundledResource *)resource;
+	if (value->free_bundled_resource_func)
+		value->free_bundled_resource_func (resource);
+}
+
+//---------------------------------------------------------------------------------------
+//
+// mono_bundled_resources_add handles bundling of many types of resources to circumvent
+// needing to find or have those resources on disk. The MonoBundledResource struct models
+// the union of information carried by all supported types of resources which are
+// enumerated in MonoBundledResourceType.
+//
+// bundled_resources:
+// A single hash table will hold all resources being bundled with the understanding
+// that all resources being bundled are uniquely named. The bundled resource is tagged
+// with the type of resource it represents, and the pointer added to this hash table
+// should fully represent a MonoBundled*Resource struct defined in `bundled-resources-internals.h
+//
+// Arguments:
+// ** resources_to_bundle - An array of pointers to `MonoBundledResource`, which details
+//     the type of MonoBundled*Resource information follows this pointer in memory.
+// len - The number of resources being added to the hash table
+//
+
+void
+mono_bundled_resources_add (MonoBundledResource **resources_to_bundle, uint32_t len)
+{
+	if (!bundled_resources)
+		bundled_resources = g_hash_table_new_full (g_str_hash, g_str_equal, NULL, mono_bundled_resources_value_destroy_func);
+
+	bool assemblyAdded, satelliteAssemblyAdded = false;
```

Comment:
Assign them explicitly, currently triggers build warning of uninitialized use of assemblyAdded on Windows build:

D:\a\_work\1\s\src\mono\mono\metadata\bundled-resources.c(105) : error C2220: the following warning is treated as an error
D:\a\_work\1\s\src\mono\mono\metadata\bundled-resources.c(105) : warning C4701: potentially uninitialized local variable 'assemblyAdded' used

Either do:

bool assemblyAdded = false, satelliteAssemblyAdded = false;

or split it into two separate lines:

bool assemblyAdded = false;
bool satelliteAssemblyAdded = false;